crates/lib3h/src/engine/real_engine.rs

```diff
@@ -101,48 +99,88 @@ impl<'engine> RealEngine<'engine> {
-            DhtConfig::with_real_engine_config(format!("{}_tId", name).as_str(), &binding, &config);
+        let mut memory_transport = GhostTransportMemory::new();
+        let mut memory_network_endpoint = Detach::new(
+            memory_transport
+                .take_parent_endpoint()
+                .expect("exists")
+                .as_context_endpoint_builder()
+                .request_id_prefix("tmem_to_child_")
+                .build::<GatewayUserData, Lib3hTrace>(),
+        );
+
+        // Bind & create this_net_peer
+        let mut gateway_ud = GatewayUserData::new();
+        let _res = memory_network_endpoint.request(
+            Lib3hTrace,
+            transport::protocol::RequestToChild::Bind {
+                spec: config.bind_url.clone(),
+            },
+            Box::new(|mut ud, response| {
+                let response = {
+                    match response {
+                        GhostCallbackData::Timeout => panic!("timeout"),
+                        GhostCallbackData::Response(response) => match response {
+                            Err(e) => panic!("{:?}", e),
+                            Ok(response) => response,
+                        },
+                    }
+                };
+                if let transport::protocol::RequestToChildResponse::Bind(bind_data) = response {
+                    ud.binding = bind_data.bound_url;
+                } else {
+                    panic!("bad response to bind: {:?}", response);
+                }
+                Ok(())
+            }),
+        );
+        let _ = memory_transport.process();
+        let _ = memory_network_endpoint.process(&mut gateway_ud);
+        let this_net_peer = PeerData {
+            peer_address: format!("{}_tId", name),
+            peer_uri: gateway_ud.binding.clone(),
+            timestamp: 0, // TODO #166
+        };
+        // Create DhtConfig
+        let dht_config = DhtConfig::with_real_engine_config(
+            &format!("{}_tId", name),
+            &gateway_ud.binding,
+            &config,
+        );
         // Create network gateway
-        let network_gateway = GatewayWrapper::new(P2pGateway::new(
-            NETWORK_GATEWAY_ID,
-            network_transport.clone(),
-            dht_factory,
-            &dht_config,
-        ));
-        debug!(
-            "New MOCK RealEngine {} -> {:?}",
-            name,
-            network_gateway.as_mut().get_this_peer_sync(),
+        let network_gateway = GatewayParentWrapperDyn::new(
+            Box::new(P2pGateway::new(
+                NETWORK_GATEWAY_ID,
+                memory_network_endpoint,
+                dht_factory,
+                &dht_config,
+            )),
+            "network_gateway_",
         );
+        debug!("New MOCK RealEngine {} -> {:?}", name, this_net_peer);
         let transport_keys = TransportKeys::new(crypto.as_crypto_system())?;
+        let multiplexer = TransportMultiplex::new(Box::new(memory_transport));
```

Comment:
multiplexer is designed to take the network_gateway as its inner_transport.

... I'm not sure how exactly to go about that... do we need to implement a de-mangler that sits on top of the gateway that only sends / receives the transport bits of the protocol?